src/dungeon_move.c

```diff
@@ -0,0 +1,3314 @@
+                    TryDisplayDungeonLoggableMessage3(attacker, currTarget, gUnknown_80F9384); // The move missed
+                }
+                else if (lightRodRedirect) {
+                    TryDisplayDungeonLoggableMessage3(attacker, currTarget, gUnknown_80F93A4); // The move missed
+                }
+                else {
+                    TryDisplayDungeonLoggableMessage3(attacker, currTarget, gUnknown_80F9364); // is unaffected!
+                }
+
+                if (sub_8045888(currTarget)) {
+                    sub_803ED30(9999, currTarget, 1, -1);
+                }
+
+                switch (moveId) {
+                    case MOVE_HI_JUMP_KICK:
+                        sub_8059FC8(attacker, currTarget, move, itemId, 1);
+                        break;
+                    case MOVE_JUMP_KICK:
+                        sub_8059E54(attacker, currTarget, move, itemId, 1);
+                        break;
+
+                }
+
+                if (sub_8044B28())
+                    break; // breaks out of the loop
+            }
+            else {
+                s32 expMultiplierBeforeMove = targetInfo->expMultiplier;
+                if (targetInfo->isNotTeamMember) {
+                    if (move->id != MOVE_REGULAR_ATTACK && itemId == 0 && expMultiplierBeforeMove == EXP_HALVED) {
+                        targetInfo->expMultiplier = EXP_REGULAR;
+                    }
+                    if (isLinkedMove == TRUE) {
+                        targetInfo->expMultiplier = EXP_BOOSTED;
+                    }
+                }
+
+                if (targetInfo->unk164 == 0xFF && sub_80717A4(currTarget, MOVE_SLEEP_TALK)) {
+                    targetInfo->unk164 = GetDirectionTowardsPosition(&currTarget->pos, &attacker->pos);
+                }
+                if (targetInfo->unk165 == 0xFF && sub_80717A4(currTarget, MOVE_SNORE)) {
+                    targetInfo->unk165 = GetDirectionTowardsPosition(&currTarget->pos, &attacker->pos);
+                }
+
+                if (HasAbility(currTarget, ABILITY_PRESSURE)
+                    && GetTreatmentBetweenMonsters(attacker, currTarget, TRUE, FALSE) == 1
+                    && move->id != MOVE_REGULAR_ATTACK
+                    && move->id != MOVE_STRUGGLE
+                    && itemId == 0
+                    && move->PP != 0)
+                {
+                    if (!(move->moveFlags2 & MOVE_FLAG2_x10)) {
+                        sub_8042950(attacker);
+                    }
+                    move->moveFlags2 |= MOVE_FLAG2_x10;
+                }
+
+                if ((GetMoveTargetAndRangeForPokemon(attacker, move, FALSE) & 0xF) == 0) {
+                    SetShopkeeperAggression(attacker, currTarget);
+                }
+                targetInfo->unk158 = 1;
+
+                // They really must've had a soft spot for all these giant switches
+                switch (moveId) {
+                    case 9:
```

Comment:
Looks like these cases could all use move ID macros. Maybe making a quick replace script would be easier with the sheer number of cases.

Reply:
Chat GPT wrote a cool script which did the job :D